### PR TITLE
GO-2261: export finished message appears before the process is done

### PR DESCRIPTION
**Pull request by AnastasiaShemyakinskaya**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### Description

Move code from old repo and remove not needed tasks for file saving

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Related Tickets & Documents

https://linear.app/anytype/issue/GO-2261/export-finished-message-appears-before-the-process-is-complete
### Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [x] 🙅 no documentation needed

### CODE REVIEW

**Review by fat-fellow — APPROVED** [review]

File: core/block/export/export.go [hunk 1]
```diff
@@ -146,22 +146,24 @@ func (e *export) Export(ctx context.Context, req pb.RpcObjectListExportRequest)
...
+		tasks := make([]process.Task, 0, len(docs))
 		for docId := range docs {
-			did := docId
-			if err = queue.Wait(func() {
-				log.With("objectID", did).Debugf("write doc")
+			var did = docId
+			f := func() {
 				if werr := e.writeDoc(ctx, req.Format, wr, docs, queue, did, req.IncludeFiles, req.IsJson); werr != nil {
-					log.With("objectID", did).Warnf("can't export doc: %v", werr)
+					log.With("threadId", did).Warnf("can't export doc: %v", werr)
```
Comment (deff7) [inline core/block/export/export.go:154]: `objectId`?

File: core/block/export/export.go [hunk 2]
```diff
@@ -146,22 +146,24 @@ func (e *export) Export(ctx context.Context, req pb.RpcObjectListExportRequest)
 				}
 			}
 		}
+		tasks := make([]process.Task, 0, len(docs))
 		for docId := range docs {
-			did := docId
-			if err = queue.Wait(func() {
-				log.With("objectID", did).Debugf("write doc")
+			var did = docId
```
Comment (deff7) [inline core/block/export/export.go:151]: `docId := docId` works too

File: core/block/export/export.go [hunk 3]
```diff
@@ -146,22 +146,24 @@ func (e *export) Export(ctx context.Context, req pb.RpcObjectListExportRequest)
...
+			f := func() {
```
Comment (deff7) [inline core/block/export/export.go:152]: maybe rename `f` to `task`?

File: core/block/export/export.go (hunk 1 above)
Comment (AnastasiaShemyakinskaya) [reply core/block/export/export.go]: Fixed

File: core/block/export/export.go (hunk 2 above)
Comment (AnastasiaShemyakinskaya) [reply core/block/export/export.go]: Fixed

File: core/block/export/export.go (hunk 3 above)
Comment (AnastasiaShemyakinskaya) [reply core/block/export/export.go]: Fixed

**Review by deff7 — APPROVED** [review]

**Pull request merged**